bundles/org.eclipse.equinox.p2.transport.ecf/src/org/eclipse/equinox/internal/p2/transport/ecf/RepositoryTransport.java

```diff
@@ -66,7 +66,9 @@ public static HttpAction of(String literal) {
 		}
 	}
 
-	private static final HttpAction HTTP_ACTION = HttpAction.of(System.getProperty("p2.httpAction")); //$NON-NLS-1$
+	private static final Map<String, ProtocolRule> RULES = Map.of( //
+			"http", ProtocolRule.of(System.getProperty("p2.httpRule")), //$NON-NLS-1$ //$NON-NLS-2$
```

Comment:
I found roughly 23 other system properties that start with "p2." including "p2.skipRepositoryProtocolCheck" in the same file.  The "eclipse.p2" prefix is also quite popular.  It's hard to say how popular "org.eclipse.equinox.p2." is as a prefix, because such strings are used for so many non-system property purposes...  In any case, I don't think there is risk of colliding accidentally with some other use case...

Reply:
I think the point is more that this is for one specific transport implementation in bundle `org.eclipse.equinox.p2.transport.ecf` and not a generic p2 property...

Reply:
I would hope that any implementation of org.eclipse.equinox.internal.p2.repository.Transport respected these properties, and that if there were ever to be more than one implementation, I would hope the logic would be refactored into a base class that is reused...

Reply:
The base class for transport implementations is

https://github.com/merks/p2/blob/563805f655d5ecad1d81afc19cb3f50eb21a82ec/bundles/org.eclipse.equinox.p2.repository/src/org/eclipse/equinox/internal/p2/repository/Transport.java

so if you want that other implementations do so (at least Tycho has a custom implementation) it should then be part of that class and the `RepositoryTransport` is calling a (static?) method there, but currently its an implementation detail of the `RepositoryTransport` class.